pilot/pkg/security/authz/model/util.go

```diff
@@ -109,3 +109,37 @@ func found(key string, list []string) bool {
 	}
 	return false
 }
+
+// getIdentityWithNewTrustDomain returns a new SPIFFE identity with the new trust domain.
+// The trust domain corresponds to the trust root of a system.
+// Refer to
+// [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+// In Istio authorization, an identity is presented in the format:
+// <trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+func getIdentityWithNewTrustDomain(trustDomain, spiffeIdentity string) string {
```

Comment:
The function should really be called `replaceTrustDomainInPrincipal`, also make the argument name consistent, you used `principal` in `extractTrustDomainFromAuthzPrincipal` then you should stick with it here, instead of using `spiffeIdentity`